src/dune.ml

```diff
@@ -0,0 +1,38 @@
+open Import
+
+type t =
+  { root : Path.t
+  ; bin : Cmd.spawn
+  }
+
+let binary = Path.of_string "dune"
+
+let make ~root () =
+  let open Promise.Syntax in
+  (* Should we do something specific for Windows ? *)
+  let spawn = { Cmd.bin = binary; args = [] } in
+  let+ spawn = Cmd.check_spawn spawn in
+  match spawn with
+  | Ok bin -> Some { bin; root }
+  | Error _ -> None
+;;
+
+let detect_dune_lock_dir t =
```

Comment:
Naming nitpick:

```suggestion
let is_project_locked t =
```

(Starting the name of a function that returns a bool with "is"/"has" or similar is quite a common naming convention. Simplifies a bit reading through the code later.)